packages/react-composites/tests/browser/chat/ChatComposite.test.ts

```diff
@@ -63,21 +70,56 @@ test.describe('Chat Composite E2E Tests', () => {
     await browser.close();
   });
 
-  test('composite loads correctly with participant list', async () => {
-    await pages[0].bringToFront();
-    await pages[0].waitForSelector(dataUiId(IDS.sendboxTextfield));
-    await pages[0].waitForSelector(dataUiId(IDS.participantList));
-    stubTimestamps(pages[0]);
-    expect(await pages[0].screenshot()).toMatchSnapshot('1-chat-screen.png');
+  test('composite pages load completely', async () => {
+    for (const idx in pages) {
+      await compositeLoaded(pages[idx]);
+      expect(await pages[idx].screenshot()).toMatchSnapshot(`1-page-${idx}-chat-screen.png`);
+    }
   });
 
   test('can send message', async () => {
+    const page = pages[0];
+    await page.bringToFront();
+    await page.type(dataUiId(IDS.sendboxTextfield), 'How the turn tables');
+    await page.keyboard.press('Enter');
+    await page.waitForSelector(`[data-ui-status="delivered"]`);
+    stubTimestamps(page);
+    expect(await page.screenshot()).toMatchSnapshot('2-send-message.png');
+  });
+
+  test('can receive message', async () => {
+    const page = pages[1];
+    await page.bringToFront();
+    await page.waitForSelector(`[data-ui-status="delivered"]`);
+    stubTimestamps(page);
+    expect(await page.screenshot()).toMatchSnapshot('3-receive-message.png');
+  });
+
+  test('read message has a viewed status', async () => {
+    const page = pages[0];
+    await page.bringToFront();
+    await page.waitForSelector(`[data-ui-status="seen"]`);
```

Comment:
I'm confused by this test. Who sent the message?

Reply:
page[0] sent the message inside test `can send message`
Any suggestions to make it more clear?

Reply:
page[0] sent the message in a previous test.
Any suggestions to make this clear?

Reply:
Covered in another discussion thread, so closing this one.

I didn't realize that these tests aren't isolated.